### DISCUSSION TITLE
JSON format documentation

### DISCUSSION THREAD

**Discussion by gussmith23** [YosysHQ/nextpnr#991] [Q&A] [answered]
I'd like to try to generate the JSON format directly. Where can I find documentation/source code for the JSON format?

**Comment by gatecat** [comment, marked as answer]
there's some basic documentation inside Yosys: https://github.com/YosysHQ/yosys/blob/63c9c9be5c0b0cc2b7f4588f1ac8e72eabc6bd0a/backends/json/json.cc#L353

do let me know if you have any more questions than that, though

**Reply by gussmith23** [reply]
Thank you so much!
